irs-edc-client/src/test/java/org/eclipse/tractusx/irs/edc/client/contract/service/EdcContractAgreementServiceTest.java

```diff
@@ -50,22 +51,23 @@ class EdcContractAgreementServiceTest {
 
     @Mock
     private RestTemplate restTemplate;
-    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    @Spy
     private EdcConfiguration edcConfiguration;
 
     private EdcContractAgreementService edcContractAgreementService;
 
     @BeforeEach
     void setUp() {
+        edcConfiguration.getControlplane().setEndpoint(new EdcConfiguration.ControlplaneConfig.EndpointConfig());
+        edcConfiguration.getControlplane().getEndpoint().setData("https://irs-consumer-controlplane.dev.demo.catena-x.net/data");
```

Comment:
technically the endpoint would be /management but I guess this does not matter for the test